code/__HELPERS/unsorted.dm

```diff
@@ -1022,6 +1022,38 @@ B --><-- A
 
 */
 
+/proc/getCone(var/turf/startPoint, var/dir, var/distance)
+	switch(dir)
+		if(1,2,4,8)
+			return getConeCardinal(startPoint, dir, distance)
+		else
+			return getConeDiagonal(startPoint, dir, distance)
+
+/proc/getConeCardinal(var/turf/startPoint, var/dir, var/distance)
+	var/up = turn(dir, 45)
+	var/down = turn(dir, -45)
+	var/list/turfs = new
+	var/turf/uppoint = startPoint
+	var/turf/downpoint = startPoint
+	turfs += startPoint
+	for(var/i=0, i<distance, i++) //Bad number for distance
+		uppoint = get_step(uppoint, up)
+		downpoint = get_step(downpoint,down)
+		turfs += block(uppoint, downpoint)
+	return turfs
+
+
+/proc/getConeDiagonal(var/turf/startPoint, var/dir, var/distance)
+	var/up = turn(dir, 45)
+	var/turf/highpoint = startPoint
+	var/list/turfs
+	for(var/i=0, i<distance, i++)
```

Comment:
That'll be off by one, remie; he uses `<` not `<=`
They want either
`for(var/i in 1 to distance)` or
`for(var/i in 0 to distance-1)`
I'd prefer the first one